src/javascript/SelectorTypes/Picker/PickerDialog/PickerDialog2.jsx

```diff
@@ -162,6 +164,17 @@ export const PickerDialog = ({
         }
     }, [dispatch, editorContext, isOpen, initialSelectedItem, pickerConfig, selectedItem, state, currentFolderInfo]);
 
+    useEffect(() => {
+        if (isOpen && inputContext.actionContext.fieldData) {
```

Comment:
can't we rely on the "initialSelectedItem" ?

Reply:
Fileddata comes from usePickerInputData hook which we have in configs, I think it should be pretty reliable. There can be an error when it is not found in which case fieldData is not set.